homework/index.js

```diff
@@ -30,15 +32,138 @@
+      createAndAppend('td', tr, {
+        text: `${detail}: `,
+        class: `label ${detail.toLowerCase()}-head`,
+      });
+      createAndAppend('td', tr, { class: `${detail.toLowerCase()}-data` });
+    });
+    const repoName = document.querySelector('.repository-data');
+    createAndAppend('a', repoName, {
+      href: hyfRepo.html_url,
+      target: '_blank',
+      text: hyfRepo.name,
+      class: 'repo-name',
+    });
+    if (hyfRepo.description === null) {
+      document.querySelector('.description-row').setAttribute('class', 'hide-description-row');
+    } else {
+      document.querySelector('.description-data').innerText = hyfRepo.description;
+    }
+    document.querySelector('.forks-data').innerText = hyfRepo.forks;
+    document.querySelector('.updated-data').innerText = new Date(
+      hyfRepo.updated_at,
+    ).toLocaleString();
+  }
+
+  // Creating the contributions part
+  async function createContributors(rightDiv, url) {
+    createAndAppend('h3', rightDiv, {
+      text: 'Contributions: ',
+      class: 'contributor-header',
+    });
+    try {
+      const contributors = await fetchJSON(url);
+      if (!(contributors && contributors.length)) {
+        createAndAppend('div', rightDiv, {
+          text: 'No contributions found.',
+          class: 'alert-error',
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const ul = createAndAppend('ul', rightDiv, {
+          class: 'contributor-list',
+        });
+        contributors.forEach(contributor => {
+          const contributorItem = createAndAppend('li', ul, {
+            class: 'contributor-item',
+          });
+          const contributorLink = createAndAppend('a', contributorItem, {
+            href: contributor.html_url,
+            target: '_blank',
+          });
+          const contributorDiv = createAndAppend('div', contributorLink, {
+            class: 'contributor',
+          });
+          createAndAppend('img', contributorDiv, {
+            src: contributor.avatar_url,
+            class: 'contributor-avatar',
+          });
+          const contributorData = createAndAppend('div', contributorDiv, {
+            class: 'contributor-data',
+          });
+          createAndAppend('div', contributorData, {
+            text: contributor.login,
+            class: 'contributor-login',
+          });
+          createAndAppend('div', contributorData, {
+            text: contributor.contributions,
+            class: 'contributor-badge',
+          });
+        });
       }
+    } catch (err) {
+      createAndAppend('div', rightDiv, {
+        text: err.message,
+        class: 'alert-error',
+      });
+    }
+  }
+
+  async function main(url) {
+    const root = document.getElementById('root');
+
+    // Creating the header and the select menu
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('h2', header, {
```

Comment:
👍 for using the `header` tag for the heading section. However, I would suggest that you use `h1` tag instead of `h2` for the header as it is more semantic and is also important for accessibility. Usually, we shouldn't have `h2` tags without any `h1` tags in the document and similarly `h3` without `h2` and so on.

Reply:
Many thanks @yash-kapila for your time, feedback and remarks 🙂.

Yes, you are right! It's corrected 👍